components/brave_wallet/browser/keyring_service.cc

```diff
@@ -2064,32 +2085,34 @@ void KeyringService::SetKeyringDerivedAccountName(
     const std::string& address,
     const std::string& name,
     KeyringService::SetKeyringDerivedAccountNameCallback callback) {
-  auto* keyring = GetHDKeyringById(keyring_id);
-  if (address.empty() || name.empty() || !keyring) {
+  if (address.empty() || name.empty()) {
     std::move(callback).Run(false);
     return;
   }
 
-  const absl::optional<size_t> index = keyring->GetAccountIndex(address);
-  if (!index) {
-    std::move(callback).Run(false);
-    return;
+  auto accounts = GetDerivedAccountsForKeyring(profile_prefs_, keyring_id);
+  for (auto& account : accounts) {
+    if (account.account_address == address) {
+      account.account_name = name;
+      SetDerivedAccountInfoForKeyring(profile_prefs_, account, keyring_id);
+      std::move(callback).Run(true);
+      NotifyAccountsChanged();
+      return;
+    }
   }
 
-  SetAccountMetaForKeyring(profile_prefs_,
-                           GetAccountPathByIndex(index.value(), keyring_id),
-                           name, address, keyring_id);
-  NotifyAccountsChanged();
-  std::move(callback).Run(true);
+  std::move(callback).Run(false);
```

Comment:
No need to go to keyring, just find account in prefs by address, rename it and update in prefs